src/lib/permissions/core/policies.ts

```diff
@@ -106,14 +106,19 @@ export function buildComputePermissionsWithPermissionFilteringPolicies<
     for (const policy of policies) {
       let hasTrueFlag = false;
 
-      const newFlags = await policy({
+      let newFlags = policy({
         flags: applicableFlags,
         resource,
         userId: request.userId,
         isAdmin: spaceRole?.isAdmin,
         preComputedSpacePermissionFlags
         // Any flag is needed here so that downstream compiler doesn't throw an error
       } as any as PermissionFilteringPolicyFnInput<R & ResourceWithSpaceId, F, true>);
+
+      if (newFlags instanceof Promise) {
```

Comment:
TBH I think you can await anything, not just promises, and it will just work

Reply:
Small performance improvement which you suggested a while ago, avoids turning everything into a promise